lib/IDE/CodeCompletion.cpp

```diff
@@ -1397,20 +1397,31 @@ bool CodeCompletionCallbacksImpl::trySolverCompletion(bool MaybeFuncBody) {
                           CurDeclContext, CompletionContext, Consumer);
     return true;
   }
-  case CompletionKind::StmtOrExpr:
+  case CompletionKind::AccessorBeginning:
   case CompletionKind::ForEachSequence:
-  case CompletionKind::PostfixExprBeginning: {
-    assert(CodeCompleteTokenExpr);
+  case CompletionKind::PostfixExprBeginning:
+  case CompletionKind::StmtOrExpr: {
     assert(CurDeclContext);
 
     ExprTypeCheckCompletionCallback Lookup(CodeCompleteTokenExpr,
                                            CurDeclContext);
-    llvm::SaveAndRestore<TypeCheckCompletionCallback *> CompletionCollector(
-        Context.CompletionCallback, &Lookup);
-    typeCheckContextAt(CurDeclContext, CompletionLoc);
-
-    if (!Lookup.gotCallback()) {
-      Lookup.fallbackTypeCheck(CurDeclContext);
+    if (CodeCompleteTokenExpr) {
+      // Happens when completing inside e.g.
```

Comment:
```suggestion
      // 'CodeCompletionTokenExpr == nullptr' happens when completing inside e.g.
```
because this is inside `if (CodeCompeltionToken)` branch.